cpp/src/toppra/algorithm/toppra.cpp

```diff
@@ -0,0 +1,25 @@
+#include <memory>
+#include <toppra/algorithm.hpp>
+#include <toppra/algorithm/toppra.hpp>
+#include <toppra/solver.hpp>
+#include <toppra/solver/qpOASES-wrapper.hpp>
+#include <toppra/toppra.hpp>
+
+namespace toppra {
+namespace algorithm {
+
+TOPPRA::TOPPRA(const LinearConstraintPtrs &constraints,
+               const GeometricPath &path)
+    : PathParametrizationAlgorithm{constraints, path} {};
+
+int TOPPRA::computePathParametrization(Vector &path_parametrization) {
+  m_solver = std::make_shared<solver::qpOASESWrapper>(
+      m_constraints, m_path,
+      Vector::LinSpaced(m_N, m_path.pathInterval()(0),
+                        m_path.pathInterval()(1)));
```

Comment:
Interestingly this dummy code does not run. I have this error. Have not really look into it though, but any idea?

```
unknown file: Failure
C++ exception with description "Invalid times." thrown in the test body.
```

Reply:
No. It doesn't ring a bell. The exception message is not very informative. Maybe it could be changed.

Reply:
#94 should fix this issue.

Reply:
I apply #94 on top of this PR and was able to run the full algorithm without any issue. I am comparing the output parametrization with the Python side and see if they are the same. If everything goes well we should have working code before very soon.